### ISSUE TITLE
Strange logic for deriving month

### ISSUE THREAD

**Issue by sfo** [python-metar/python-metar#106]
Consider this code:

https://github.com/python-metar/python-metar/blob/dd4dab1d3c8befd462baac0f38259dd5ee3ca7d7/metar/Metar.py#L582-L588

This gives currently (beginning of March) the following error: `ValueError: day is out of range for month`, e.g. for this record: `METAR EDDM 300020Z 06004KT 9999 2000NE BCFG SKC 04/03 Q1005`

Because the record's day (30) is greater than today (5), the current month (March / 3) is set to February (2), which has only 29 days, resulting in the error.

I've got two suggestions, one of them would break the API:
1) make month and year a mandatory parameter.
2) if the idea of guessing the month should be retained, always guess a month that has 31 days.

**Comment by tomp** [comment]
This logic works correctly if you're processing current reports, which is the only context in which it's expected to work.  If you're processing historical data, you should, of course, specify the year and the month explicitly, but I don't think that would justify forcing the year and month to always be provided.

Maybe this could be handled more gracefully by having the exception suggest that the year and month need to be provided, when an error like this is detected?  That would probably be less confusing than the ValueError that's raised now, and seems better than silently replacing one wrong guess with another.

**Comment by sfo** [comment]
Yeah, I agree, this would be option number three. But this still leaves us with the problem of inconsistent handling of .

If the library is defined to process "current" (whatever this means?) reports only, every other report that does not fit should raise the error and guessing should be eliminated completely, because no matter what, there is no possibility to get this one right due to inherently missing information.

But that raises the question, how "current" is defined. Frankly spoken, every report that was not (presumably) issued within the last 30 minutes would have to be discarded.

**Comment by akrherz** [comment]
Thanks for the comments here.  I am happy with the current approach and the only change I'd support is to specify a full `datetime` instance instead of just a `year` and `month`.

PS: see related discussion here Unidata/MetPy#1327

**Comment by tomp** [comment]
For what it's worth, I think of "current" reports as being anything within a week of today.  I'm thinking of someone who's trying to ingest reports from the last few hours, and perhaps the last few days. 
The first serious user of this module was an environmental engineer in Seattle, who was processing all global METAR reports every hour, every hour, every day for his work.  A number of design decisions and optimizations were made to support that, and I think this was one of them.

Daryl's suggestion is fine with me.
